File: src/FluentValidationCLI.Frontend/src/validators/UserValidator.ts
import { Validator } from 'fluentvalidation-ts';

export interface User {
  username: string;
  email: string;
  age: number;
}

export class UserValidator extends Validator<User> {
  constructor() {
    super();

    this.ruleFor('username').notEmpty().withMessage('Username is required').length(3, 20).withMessage('Username must be between 3 and 20 chars');
    this.ruleFor('email').notEmpty().emailAddress();
    this.ruleFor('age').greaterThan(18).lessThan(100);
  }
}

File: src/FluentValidationCLI.Backend/Validators/UserValidator.cs
using FluentValidation;
using FluentValidationCLI.Backend.Models;

namespace FluentValidationCLI.Backend.Validators;

public class UserValidator : AbstractValidator<User>
{
    public UserValidator()
    {
        RuleFor(x => x.Username).NotEmpty().WithMessage("Username is required").Length(3, 20).WithMessage("Username must be between 3 and 20 chars");
        RuleFor(x => x.Email).NotEmpty().EmailAddress();
        RuleFor(x => x.Age).GreaterThan(18).LessThan(100);
    }
}
